### PR TITLE
don't show shell integration tooltip for certain terminals 

### LINKED ISSUE #151163: Shell integration failed to activate message appears on hover for tasks
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.68-insider
- OS Version: Windows 11

Steps to Reproduce:

1. Run a bunch of build tasks
2. Hover on one of the tasks
3. :bug: see a message about shell integration failing to activate 
image

### PR DESCRIPTION
**Pull request by meganrogge**
This PR fixes #151163

### CODE REVIEW

**Review by Tyriar — DISMISSED** [review]
The name isn't the best now if we're using it for that, maybe `disableShellIntegrationStatus`, `shellIntegrationInjected`?

**Comment by meganrogge** [comment]
maybe `disableShellIntegrationReporting`?

File: src/vs/workbench/contrib/terminal/browser/terminal.ts [hunk 1]
```diff
@@ -458,6 +458,11 @@ export interface ITerminalInstance {
 	 */
 	target?: TerminalLocation;
 
+	/**
+	 * Whether or not shell integration telemetry should be recorded for this terminal.
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminal.ts:462]:
```suggestion
	 * Whether or not shell integration telemetry / warnings should be reported for this terminal.
```

**Comment by Tyriar** [comment]
Reporting fixes the problem👍

We can tweak it later to maybe be a status enum or something with more than 2 states if needed

**Review by Tyriar — APPROVED** [review]

**Pull request merged**